pkg/config/generate.go

```diff
@@ -32,15 +33,10 @@ func createUserHtPassEntry(name string, passwd *string, htPasswd []byte) ([]byte
 	return htPassEntry, nil
 }
 
-func (g *simpleGenerator) Generate(cs *api.OpenShiftManagedCluster) (err error) {
+func (g *simpleGenerator) Generate(cs *api.OpenShiftManagedCluster, template *pluginapi.Config) (err error) {
+	cs.Config = api.ConvertFromPlugin(template, &cs.Config)
```

Comment:
&cs.Config is never going to be nil. It's always a pointer to an empty struct in case of a create.